### PR TITLE
Add list Clusters to AKS UI proxy

### PR DESCRIPTION
**Pull request by mjura**

### CODE REVIEW

**Review by thedadams — COMMENTED** [review]
Looks good, but 32454 should be merged first.

**Review by cmurphy — COMMENTED** [review]
lgtm after https://github.com/rancher/rancher/pull/32454 is merged

File: go.sum [hunk 1]
```diff
@@ -51,32 +51,48 @@ github.com/Azure/go-autorest v14.2.0+incompatible/go.mod h1:r+4oMnoxhatjLLJ6zxSW
 github.com/Azure/go-autorest/autorest v0.9.0/go.mod h1:xyHB1BMZT0cuDHU7I0+g046+BFDTQ8rEZB0s4Yfa6bI=
 github.com/Azure/go-autorest/autorest v0.9.3/go.mod h1:GsRuLYvwzLjjjRoWEIyMUaYq8GNUx2nRB378IPt/1p0=
 github.com/Azure/go-autorest/autorest v0.11.12/go.mod h1:eipySxLmqSyC5s5k1CLupqet0PSENBEDP93LQ9a8QYw=
+github.com/Azure/go-autorest/autorest v0.11.12/go.mod h1:eipySxLmqSyC5s5k1CLupqet0PSENBEDP93LQ9a8QYw=
```
Comment (thedadams) [inline go.sum:54]: `go.sum` changes without `go.mod` changes?

File: pkg/api/norman/customization/aks/handler.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+	TenantID         string `json:"tenantId"`
+	SubscriptionID   string `json:"subscriptionId"`
+	ClientID         string `json:"clientId"`
+	ClientSecret     string `json:"clientSecret"`
+	ResourceLocation string `json:"region"`
+}
+
+// AKS handler lists available resources in Azure API
+type handler struct {
+	Action        string
```
Comment (thedadams) [inline pkg/api/norman/customization/aks/handler.go:43]: Is this needed?

**Review by thedadams — COMMENTED** [review]
Assuming this is mostly the same as https://github.com/rancher/rancher/pull/32454

File: pkg/api/norman/customization/aks/handler.go (hunk 2 above)
Comment (mjura) [reply pkg/api/norman/customization/aks/handler.go]: fixed

File: go.sum (hunk 1 above)
Comment (mjura) [reply go.sum:54]: fixed

**Comment by cbron** [comment]
Issue: #32587

**Review by cmurphy — APPROVED** [review]

File: pkg/api/norman/customization/aks/listers.go [hunk 3]
```diff
@@ -178,6 +191,45 @@ func listVirtualNetworks(ctx context.Context, cap *Capabilities) ([]byte, int, e
...
+
+	var clusters []clustersResponseBody
+
+	for clusterList.NotDone() {
+		for _, cluster := range clusterList.Values() {
+			var tmpCluster clustersResponseBody
+			if cluster.ID != nil {
+				tmpCluster.ResourceGroup = matchResourceGroup.FindStringSubmatch(to.String(cluster.ID))[1]
+			}
+			if cluster.Name != nil {
```
Comment (thedadams) [inline pkg/api/norman/customization/aks/listers.go:218]: nit: this check is not needed.

**Review by thedadams — APPROVED** [review]

File: pkg/api/norman/customization/aks/listers.go [hunk 4]
```diff
@@ -26,7 +27,7 @@ type subnet struct {
 	AddressRange string `json:"addressRange"`
 }
 
-var matchAzureNetwork = regexp.MustCompile("/resourceGroups/(.+?)/")
+var matchResourceGroup = regexp.MustCompile("/resource[gG]roups/(.+?)/")
```
Comment (cbron) [inline pkg/api/norman/customization/aks/listers.go:30]: Curious what forced this change ?

File: pkg/api/norman/customization/aks/listers.go [hunk 5]
```diff
@@ -178,6 +191,45 @@ func listVirtualNetworks(ctx context.Context, cap *Capabilities) ([]byte, int, e
...
+		return nil, http.StatusBadRequest, fmt.Errorf("failed to get cluster list: %v", err)
+	}
+
+	var clusters []clustersResponseBody
+
+	for clusterList.NotDone() {
+		for _, cluster := range clusterList.Values() {
+			var tmpCluster clustersResponseBody
+			if cluster.ID != nil {
+				tmpCluster.ResourceGroup = matchResourceGroup.FindStringSubmatch(to.String(cluster.ID))[1]
```
Comment (cbron) [inline pkg/api/norman/customization/aks/listers.go:216]: How do you know you will find a match and not hit `index out of range` error ? I think you should check the length first.

**Pull request merged**